### PR TITLE
storageService.create_file: raise HTTP errors, refactor

### LINKED ISSUE #815: Problem: failure in "Store the AIP" not reflected in the GUI
When the storeAIP_v0.0 micro-service runs and makes an API call to the SS to store the AIP and the SS returns a bad status code, the micro-service gives no indication to the user that the AIP storage has failed. Often the AIP is in fact stored but it may not have been stored correctly and completely (e.g., a pointer file may not have been created). AIP storage should probably occur in a transaction of sorts and should be rolled back if not completed fully and correctly (which is an issue for the SS).

Related to https://github.com/artefactual/archivematica/issues/809.

### PR DESCRIPTION
**Pull request by hakamine**
Fixes #815 
Note that this only fixes the issue of the "store AIP" job/micro-service not failing if getting an error from the SS (probably also need to roll back changes in the SS database in another PR)

### CODE REVIEW

File: src/archivematicaCommon/lib/storageService.py [hunk 1]
```diff
@@ -285,11 +285,11 @@ def create_file(uuid, origin_location, origin_path, current_location,
         else:
             url = _storage_service_url() + 'file/'
             response = session.post(url, json=new_file)
-    except requests.exceptions.RequestException as e:
+        # raise an HTTPError exception if getting HTTP 4xx or 5xx error
+        response.raise_for_status()
+    except (requests.exceptions.RequestException, requests.exceptions.HTTPError) as e:
```
Comment (sevein) [inline src/archivematicaCommon/lib/storageService.py]: `requests.RequestException` is the base class in Request library so we're already catching all the subclasses too. See https://github.com/requests/requests/blob/master/requests/exceptions.py.

Comment (sevein) [inline src/archivematicaCommon/lib/storageService.py]: `requests.RequestException` is the base class in Request library so we're already catching all the subclasses too. See https://github.com/requests/requests/blob/master/requests/exceptions.py.

File: src/archivematicaCommon/lib/storageService.py [hunk 2]
```diff
@@ -285,11 +285,11 @@ def create_file(uuid, origin_location, origin_path, current_location,
         else:
             url = _storage_service_url() + 'file/'
             response = session.post(url, json=new_file)
-    except requests.exceptions.RequestException as e:
+        # raise an HTTPError exception if getting HTTP 4xx or 5xx error
+        response.raise_for_status()
```
Comment (sevein) [inline src/archivematicaCommon/lib/storageService.py]: 👍

File: src/archivematicaCommon/lib/storageService.py [hunk 3]
```diff
@@ -285,11 +285,11 @@ def create_file(uuid, origin_location, origin_path, current_location,
...
+    except (requests.exceptions.RequestException, requests.exceptions.HTTPError) as e:
         LOGGER.warning("Unable to create file from %s because %s", new_file, e)
         return (None, e)
```
Comment (sevein) [inline src/archivematicaCommon/lib/storageService.py]: FYI I think that at some point we should change the way this function returns - it happens with other functions in this module. We shouldn't be returning the error, the idiomatic way in Python is to return the expected value or raise an exception instead. Pretty much like requests is doing too! An exception can carry information about the error too. To be clear, I'm not asking to change it - just saying! :)

**Review by sevein — CHANGES_REQUESTED** [review]
Thanks for doing this, @hakamine.

**Comment by hakamine** [comment]
Thank you @sevein. As suggested, create_file has been refactored to return the expected value or raise an exception instead (also parts of the code that call this function have been fixed accordingly). Did a quick test storing to aipstore and backlog, and it seems working fine.

**Comment by sevein** [comment]
@hakamine thanks, I've added a couple of fixes: I changed the signature of a return we left behind in the create_file function and an extra commit that it's more of a cosmetic change where we avoid the `else` and its nesting if the `except` is a catch-all.

**Comment by hakamine** [comment]
Thank you @sevein!  Would it be ok if I squash and merge?

**Review by sevein — APPROVED** [review]

**Pull request merged**